### PR TITLE
Fix channel-details not shown

### LINKED ISSUE #1784: Channel Details is broken
## The problem

When clicking "Channel Details":
image

I see this:
image

## Expected behavior

Channel details should be shown, i.e. Channel UID.

## Steps to reproduce

Navigate to a random thing and a random channel and click "Channel Details".

## Your environment

```yaml
runtimeInfo:
  version: 4.0.0
  buildString: "Build #3358"
locale: da-DK
systemInfo:
  configFolder: C:\PROGRA~1\openHAB\conf
  userdataFolder: C:\PROGRA~1\openHAB\userdata
  logFolder: C:\PROGRA~1\openHAB\userdata\logs
  javaVersion: 17.0.5
  javaVendor: Oracle Corporation
  osName: Windows 10
  osVersion: "10.0"
  osArchitecture: amd64
  availableProcessors: 4
  freeMemory: 144139736
  totalMemory: 266338304
  startLevel: 100
bindings: null
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1
    prefersColorScheme: light
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: 8
    hardwareConcurrency: 4
    language: da
    languages:
      - da
      - en-US
      - da-DK
      - en-DK
      - en
    onLine: true
    platform: Win32
  screen:
    width: 3840
    height: 1600
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
    like Gecko) Chrome/110.0.0.0 Safari/537.36
timestamp: 2023-03-09T16:03:47.964Z
```

## Browser console

image

```txt
Failed to load resource: the server responded with a status of 404 (Not Found)
http://localhost:8080/rest/config-descriptions/channel:unifi:wirelessClient:4921f92893:169dc7c8:rssi
```

When I access this URL directly:

```json
{"error":{"message":"Authentication required","http-code":401}}
```

The same happens in 3.4.2 actually, but here the UI still works.

# Additional information

See https://community.openhab.org/t/openhab-4-0-snapshot-discussion/142322/239

Might be a regression of #1683.

### PR DESCRIPTION
**Pull request by J-N-K**
Fixes #1784.

### CODE REVIEW

**Comment by ghys** [comment]
I'll let @florian-h05 merge as I'm just catching up and not part of the conversation.

**Review by florian-h05 — COMMENTED** [review]
Instead of having `this.ready = true` both in the then and catch of the promise, we could put it behing it.
I have also noticed that the `if (err === 404) ...` doesn't catch on 404, which caused the „Channel has no configuration“ footer to not be displayed.

I will fix these two and merge then.

**Review by florian-h05 — APPROVED** [review]
LGTM, thanks!
I have tested it, it should fix the issue so I will change the PR description to fixes...

**Comment by florian-h05** [comment]
@ghys 
FYI: I have some minor improvements for channel details/edit/add I just made while having a look at this, will open a new PR tomorrow.

File: bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue [hunk 1]
```diff
@@ -55,12 +55,12 @@ export default {
...
       this.$oh.api.get(`/rest/config-descriptions/channel:${this.thing.UID}:${this.channelId.replace('#', '%23')}`).then((ct) => {
         this.configDescription = ct
-        this.ready = true
       }).catch((err) => {
-        if (err === 404) {
+        if (err === 'Not Found' || err === 404) {
           this.noConfig = true
         }
       })
+      this.ready = true
```
Comment (J-N-K) [inline bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue:63]: Isn‘t this wrong? It sets ready when the request is made, not when it is done.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue:63]: I‘ve asked my this question as well, and the answer is not easy:
Whilst I agree that it is set when the request is made, I don‘t think that this really matters. `ready` is only used by the `channel-general-settings` component, which does not need any data that is requested in this component here (all data that component needs is passed to this component and not fetched here). Theoretically, IMO we don‘t need the `ready` variable at all, we could just set the `channel-general-settings`‘s ready prop to true.

What do you think?

Comment (ghys) [reply bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue:63]: I used the `ready` flag in my implementations mostly to replace "skeletons" (https://v5.framework7.io/vue/skeleton#examples) with actual data when it was available.

What determines when a page is "ready" to be displayed is up to the developer's choice, there might be additional non-critical requests that might be made after the page is considered "ready".

In that case I'll have to check whether it's a good thing to consider the component "ready" even if the request fails.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue:63]: Let‘s discuss this in my upcoming enhancement PR, and set ready in this one as usual to get this merged and be able to publish the milestone release.

Comment (J-N-K) [reply bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue:63]: I agree.

Comment (florian-h05) [reply bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue:63]: Great, I will change this and merge then.

**Comment by jlaur** [comment]
@J-N-K or @florian-h05 - can you add the "regression" label, so it won't appear in the release notes?

**Comment by florian-h05** [comment]
Done

**Pull request merged**